cluster/cluster.go

```diff
@@ -0,0 +1,373 @@
+	default:
+		return []string{"/mnt/disk1", "/mnt/disk2", "/mnt/disk3", "/mnt/disk4"}
+	}
+
+}
+
+const MaxNumberDiskPerNode = 4
+const MaxNumberHost = 4
+
+//Creates a service that will resolve to any of the hosts within the storage cluster this tenant lives in
+func CreateDeploymentWithTenants(tenants []Tenant, storageClusterNum string, hostNum string) {
+	config := getConfig()
+	// creates the clientset
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		panic(err.Error())
+	}
+
+	scHostName := fmt.Sprintf("sc-%s-host-%s", storageClusterNum, hostNum)
+	var replicas int32 = 1
+
+	mainPodSpec := v1.PodSpec{
+		NodeSelector: map[string]string{
+			"kubernetes.io/hostname": nodeNameForSCHostNum(storageClusterNum, hostNum),
+		},
+	}
+
+	for i := range tenants {
+		tenant := tenants[i]
+		volumeMounts := []v1.VolumeMount{}
+		tenantContainer := v1.Container{
+			Name:            fmt.Sprintf("%s-minio-%s", tenant.Name, hostNum),
+			Image:           "minio/minio:RELEASE.2019-09-26T19-42-35Z",
+			ImagePullPolicy: "IfNotPresent",
+			Args: []string{
+				"server",
+				"--address",
+				fmt.Sprintf(":%d", tenant.Port),
+				fmt.Sprintf(
+					"http://sc-%s-host-{1...%d}:%d/mnt/tdisk{1...%d}",
+					storageClusterNum,
+					MaxNumberHost,
+					tenant.Port,
+					MaxNumberDiskPerNode),
+			},
+			Ports: []v1.ContainerPort{
+				{
+					Name:          "http",
+					ContainerPort: tenant.Port,
+				},
+			},
+			EnvFrom: []v1.EnvFromSource{
+				{
+					ConfigMapRef: &v1.ConfigMapEnvSource{
+						LocalObjectReference: v1.LocalObjectReference{Name: fmt.Sprintf("%s-env", tenant.Name)},
+					},
+				},
+			},
+		}
+		//volumes that will be used by this tenant
+		for vi := 1; vi <= MaxNumberDiskPerNode; vi++ {
+			vname := fmt.Sprintf("%s-pv-%d", tenant.Name, vi)
+			volumenSource := v1.VolumeSource{HostPath: &v1.HostPathVolumeSource{Path: fmt.Sprintf("/mnt/disk%d/%s", vi, tenant.Name)}}
+			hostPathVolume := v1.Volume{Name: vname, VolumeSource: volumenSource}
+			mainPodSpec.Volumes = append(mainPodSpec.Volumes, hostPathVolume)
+
+			mount := v1.VolumeMount{
+				Name:      vname,
+				MountPath: fmt.Sprintf("/mnt/tdisk%d", vi),
+			}
+			volumeMounts = append(volumeMounts, mount)
+		}
+		tenantContainer.VolumeMounts = volumeMounts
+
+		mainPodSpec.Containers = append(mainPodSpec.Containers, tenantContainer)
+	}
+
+	deployment := v1beta1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: scHostName,
+		},
+		Spec: v1beta1.DeploymentSpec{
+			Replicas: &replicas,
+			Template: v1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						"app": scHostName,
+						"sc":  fmt.Sprintf("storage-cluster-%s", storageClusterNum),
+					},
+				},
+				Spec: mainPodSpec,
+			},
+		},
+	}
+
+	res, err := clientset.ExtensionsV1beta1().Deployments("default").Create(&deployment);
+	if err != nil {
+		panic(err.Error())
+	}
+	fmt.Println("done creating storage cluster deploymenty ")
```

Comment:
Is there a kubernetes specific logging helper we can use ?

Reply:
anything that goes to std/out becomes a log which can be collected by k8s, these log lines are temporary